### ISSUE TITLE
Export interface

### ISSUE THREAD

**Issue by lchimaru** [BetterCallSky/ts-mongoose#3]
Is there any chance to export created interface to be reused somewhere in application? I mean, consider following use case:  controller has basic getOne() method, which suppose to have returned type - in good day scenario it will be such interface extended by Document (which is done by ts-mongoose). Any workaround to prevent code duplication?

**Comment by BetterCallSky** [comment]
You can use conditional types from typescript

```ts
import { Model } from 'mongoose';
import { typedModel } from 'ts-mongoose';

type ExtractDoc<T> = T extends Model<infer U> ? U : never;

export const User = typedModel('User', UserSchema);

export type UserDoc = ExtractDoc<typeof User>;
```

**Comment by MilosStanic** [comment]
This is very important and this should be incorporated in the docs on the front page. 
I almost gave up on ts-mongoose when I found this. 
I'm using ts-mongoose with nest.js

**Comment by BetterCallSky** [comment]
Available in 0.0.12

**Comment by lchimaru** [comment]
Woaaaah, this is amazing :) Thanks.

Any btw for others looking for ExtractedDoc to be split into pure interface without Document fields just use below utility function:
`export type ExtractInterfaceFromDoc<T> = Pick<T, Exclude<keyof T, keyof Document>>;`

**Comment by michaelbats** [comment]
@BetterCallSky  what do you mean available in 0.0.12? is there something included in the package that bypasses the need for having that export type?

Also @lchimaru  I'm not sure how your code snipped is supposed to work and how it fits in the example provided above :D 

I'm trying to get the interface from the document without the Document stuff.

**Comment by lchimaru** [comment]
@michaelbats There is `ExtractProps` helper to get Schema props without Document stuff. I was added in 0.0.13.
Look at below example:
```ts
export const UserSchema = createSchema({
  email: Type.string(),
  username: Type.string(),
  isBlocked: Type.optionalBoolean(),
});

export type UserProps = ExtractProps<typeof UserSchema>;
```
`UserProps` will return following type:
```ts
{
  email: string,
  username: string,
  isBlocked?: boolean
}
```
